blog/README.md

```diff
@@ -0,0 +1,232 @@
+  CC [M]  /build/simple-kmod/simple-procfs-kmod.o
+  Building modules, stage 2.
+  MODPOST 2 modules
+  CC      /build/simple-kmod/simple-kmod.mod.o
+  LD [M]  /build/simple-kmod/simple-kmod.ko
+  CC      /build/simple-kmod/simple-procfs-kmod.mod.o
+  LD [M]  /build/simple-kmod/simple-procfs-kmod.ko
+make[1]: Leaving directory '/usr/src/kernels/4.18.0-372.53.1.el8_6.x86_64'
+gcc -o spkut ./simple-procfs-kmod-userspace-tool.c
+install -v -m 755 spkut /bin/
+'spkut' -> '/bin/spkut'
+install -v -m 755 -d /lib/modules/4.18.0-372.53.1.el8_6.x86_64/
+install -v -m 644 simple-kmod.ko        /lib/modules/4.18.0-372.53.1.el8_6.x86_64/simple-kmod.ko
+'simple-kmod.ko' -> '/lib/modules/4.18.0-372.53.1.el8_6.x86_64/simple-kmod.ko'
+install -v -m 644 simple-procfs-kmod.ko /lib/modules/4.18.0-372.53.1.el8_6.x86_64/simple-procfs-kmod.ko
+'simple-procfs-kmod.ko' -> '/lib/modules/4.18.0-372.53.1.el8_6.x86_64/simple-procfs-kmod.ko'
+depmod -F /lib/modules/4.18.0-372.53.1.el8_6.x86_64/System.map 4.18.0-372.53.1.el8_6.x86_64
+--> 26dfedc51be
+[2/2] STEP 1/5: FROM quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:127670aaa6073e97bd6d919e1b57bd16b60435410b096aeefce7cd64f6c83d24
+[2/2] STEP 2/5: ARG KERNEL_VERSION
+--> 2879805a870
+[2/2] STEP 3/5: COPY --from=builder /etc/driver-toolkit-release.json /etc/
+--> 8dea5bb2bd8
+[2/2] STEP 4/5: COPY --from=builder /lib/modules/4.18.0-372.53.1.el8_6.x86_64/*.ko /usr/lib/modules/4.18.0-372.53.1.el8_6.x86_64/
+--> 6249995c281
+[2/2] STEP 5/5: RUN depmod -a 4.18.0-372.53.1.el8_6.x86_64 && echo simple_kmod > /etc/modules-load.d/simple_kmod.conf
+[2/2] COMMIT quay.io/ybettan/coreos-layering:simple-kmod
+--> 612c8c8a54f
+Successfully tagged quay.io/ybettan/coreos-layering:simple-kmod
+612c8c8a54f0800cf8dcb4c20d5d3d384dc47463711d6d3ad1bd93d52e982b1e
+
+```
+
+Now, let's push the image to some image registry in order for the image to be accesible
+by the cluster.
+```
+$ podman push quay.io/ybettan/coreos-layering:simple-kmod
+```
+
+Make sure the image is public or it have the correct pull-secret to pull the image.
+The pull secret in the cluster is
+```
+$ oc get secret/pull-secret -n openshift-config.
+```
+
+### Loading the kernel modules into the nodes
+
+First, we will need to create a `MachineConfig` file
+```
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 99-ybettan-external-image
+spec:
+  osImageURL: quay.io/ybettan/coreos-layering:simple-kmod
+```
+
+Make sure to change the `osImageURL` section in the MachineConfig yaml to the
+image you just built, then apply it.
+```
+$ oc apply -f machineconfig.yaml
+```
+
+The [machine-config-operator](https://github.com/openshift/machine-config-operator)
+will start updating workers nodes one by one with the new image specified in
+`osImageURL` which contain our kernel module.
+
+This may take some time and you can track the progress using
+```
+$ watch oc get nodes
```

Comment:
You can also do that with `oc get mcp`

```
# oc get mcp
NAME     CONFIG                                             UPDATED   UPDATING   DEGRADED   MACHINECOUNT   READYMACHINECOUNT   UPDATEDMACHINECOUNT   DEGRADEDMACHINECOUNT   AGE
master   rendered-master-82eca8706478931ea5d536110b96febd   True      False      False      3              3                   3                     0                      44d
worker   rendered-worker-cb36aa3bb504e15d730a5a553e1d8ac9   True      False      False      2              2                   2                     0                      44d
```

and see which nodes are ready/updating/updated

Reply:
Do you think it is clearer than https://github.com/ybettan/coreos-layering-driver-toolkit/pull/4/files#diff-78f236b87da98227ae670c36caae7662a3f9e664737f5a8feecdbd5d8b71545aR210-R216?